### PR TITLE
hack/local-up-cluster.sh defaults to allow swap

### PR DESCRIPTION
**Pull request by derekwaynecarr**
**What this PR does / why we need it**:
developers on linux typically have swap on while developing.
defaults local-up-cluster experience to not fail kubelet if swap is enabled.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
@derekwaynecarr seen this PR? https://github.com/kubernetes/kubernetes/pull/50377

**Comment by cblecker** [comment]
It was suggested here (https://github.com/kubernetes/kubernetes/pull/50377#issuecomment-321422026) that we detect and warn. This looks good in adding an option, but it should default to true and warn so that devs can opt-in.

**Comment by luxas** [comment]
/lgtm

@timothysc faced this as well I think

**Comment by luxas** [comment]
/lgtm cancel
Oh, I missed the comments in thread above, let's wait for more ppl

**Comment by derekwaynecarr** [comment]
My humble opinion is a normal developer does not disable swap on their
local machine.  Hack local up targets developer use cases.  I think this is
the appropriate default for this context.

Another way of saying this is that community experience discussing this
issue validates this perspective.

**Comment by dims** [comment]
I buy that argument @derekwaynecarr :)

**Comment by cblecker** [comment]
What about a warning of some kind that swap is on and that's a non-default behaviour? We don't want to slow down contributors, but I see the concern about having default behaviours in local-up-cluster that don't match what the kubelet will see in the real world.

cc @liggitt who raised that concern in the other PR

**Comment by liggitt** [comment]
warning and allowing it is fine

**Comment by derekwaynecarr** [comment]
updated with warning text.

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -112,6 +114,11 @@ if [ "${CLOUD_PROVIDER}" == "openstack" ]; then
     fi
 fi
 
+# warn if users are running with swap allowed
+if [ "${FAIL_SWAP_ON}" == "false" ]; then
+    echo "WARNING : The kubelet is configured to not fail if swap is enabled; production deployments should disable swap." 2>&1
```
Comment (cblecker) [inline hack/local-up-cluster.sh]: Did you mean to redirect stderr to stdout? Or was this meant to be the other way?

Comment (derekwaynecarr) [reply hack/local-up-cluster.sh]: I missed that when copy/paste from another echo stmt.  will update to just do a normal echo.

**Comment by derekwaynecarr** [comment]
ok, fixed up.

**Pull request merged**